### ISSUE TITLE
Missing Resource IDs

### ISSUE THREAD

**Issue by awushensky** [kezong/fat-aar-android#209]
Thanks again for making this library, it has been instrumental in releasing a minified version of my library with multiple modules.

Unfortunately, with version `1.2.19`, I'm getting the following problem:

```
     Caused by: android.content.res.Resources$NotFoundException: Resource ID #0x0
        at android.content.res.ResourcesImpl.getValue(ResourcesImpl.java:237)
        at android.content.res.Resources.getColor(Resources.java:1059)
        at android.content.Context.getColor(Context.java:677)
        at androidx.core.content.ContextCompat.getColor(ContextCompat.java:517)
        at com.getbouncer.scan.ui.util.ViewExtensionsKt.getColorByRes(ViewExtensions.kt:82)
        at com.example.ExampleActivity.setupProcessingOverlayViewUi(Unknown Source:5)
```

I noticed in the `aar_rebundle/release` directory, the `R.txt` file lists many resources with ID `0x0`.

In this case, the file looks like this:
```
...
int color bouncerDebugLowConfidence 0x0
int color bouncerDebugMediumConfidence 0x0
int color bouncerEnterCardManuallyColorDark 0x0
int color bouncerEnterCardManuallyColorLight 0x0
int color bouncerFlashButtonDarkColor 0x0
int color bouncerFlashButtonLightColor 0x0
int color bouncerFoundBackground 0x0
int color bouncerFoundOutline 0x0
int color bouncerInstructionsColorDark 0x0
int color bouncerInstructionsColorLight 0x0
int color bouncerNotFoundBackground 0x0
int color bouncerNotFoundOutline 0x0
int color bouncerProcessingBackground 0x0
int color bouncerProcessingText 0x0
int color bouncerSecurityColorDark 0x0
int color bouncerSecurityColorLight 0x0
int color bouncerWrongBackground 0x0
...
```

`bouncerProcessingBackground` is defined in the same module that it is referenced, but results in an ID 0x0.

this persists after a fresh repository pull.

**Comment by awushensky** [comment]
Confirmed this is still a problem in 1.2.20

**Comment by awushensky** [comment]
@kezong this problem appears to happen only when gradle is version 4.1.0 or above. Gradle version 4.0.2 with fat-aar 1.2.20 works correctly.

**Comment by kezong** [comment]
I can't repro it locally. Can you provide a demo that can reproduce this problem?

**Comment by awushensky** [comment]
I spent some time today going through our gradle scripts and doing a lot of cleanup. It appears that one of the changes I made has resolved this problem. For posterity, the changes I made were:

- remove the `kotlin-android-extensions` plugin and replace with `buildFeatures { viewBinding true }` to perform view binding
- replacing the `kotlinx:kotlinx-serialization-runtime` dependency with `kotlinx:kotlinx-serialization-json`
- upgrading the gradle plugin to `1.4.20` from `1.4.10`
- adding `kotlin-parcelize` plugin
- replace the gradle wrapper `gradle-6.5-bin` with `gradle-6.5-all`   (this feels like the most likely candidate to me)

Thanks again for the great library @kezong , it's very helpful for distributing my SDK!
